test/test_util_encode-params.ts

```diff
@@ -79,4 +79,35 @@ describe("buildQueryString function", () => {
 
     done();
   });
+
+  it("handles strange characters correctly on param values", done => {
+    const params = {
+      filter: {
+        test: 'ñaka',
+        garçon: 'test,value,french',
+        quality: 'check-in',
+        parenthesis: 'this(null)[0]',
+        percentage: '100%'
+      }
+    };
+
+    const qs = buildQueryString(params);
+
+    qs.should.include(butEncoded('filter%5Btest%5D=%C3%B1aka'));
+    qs.should.include(butEncoded('filter%5Bgar%C3%A7on%5D=test%2Cvalue%2Cfrench'));
+    qs.should.include(butEncoded('filter%5Bquality%5D=check-in'));
+    qs.should.include(butEncoded('filter%5Bparenthesis%5D=this(null)%5B0%5D'));
+    qs.should.include(butEncoded('filter%5Bpercentage%5D=100%'));
```

Comment:
`butEncoded()` is there to make this code legible, you should use it like this:

```
qs.should.include(butEncoded('filter[test]=%C3%B1aka'));
```

If you want you can make the function to accept other unescaped characters. I didn't want to complicate my life and just used replace functions but we can elaborate it a bit more…

Reply:
Oh, well, I just wanted to check the encoding was correct, so maybe what I should do is remove the `butEncoded` method altogether.

Reply:
If the purpose of the test is to check the encoding, better not play with it and just test it. `butEncoded` is good to test the structure of the query string.